qa/full-cluster-restart/src/test/java/org/elasticsearch/upgrades/FullClusterRestartIT.java

```diff
@@ -939,9 +941,10 @@ public void testSnapshotRestore() throws IOException {
         createSnapshot.setJsonEntity("{\"indices\": \"" + index + "\"}");
         client().performRequest(createSnapshot);
 
-        checkSnapshot("old_snap", count, getOldClusterVersion());
+        checkSnapshot("old_snap", count, oldClusterVersion, type);
         if (false == isRunningAgainstOldCluster()) {
-            checkSnapshot("new_snap", count, Version.CURRENT);
+            // we always expect the "_doc" type on 7.x clusters, even if the original pre-6.7 clusters type was "doc"
```

Comment:
I'm a little confused about this statement -- if a template is added with a custom type name, then on 7.x we should still return that custom type name if `include_type_name=true`. Would you be able to explain this fix in a bit more detail?

Reply:
This is the case when we are using "doc" in the old cluster (will e.g. happen on full restart from 6.6, see setType()). This part of the check creates a new snapshot on a _new_ cluster (>7.0) and has the "doc" in the template. To me it seems this gets converted in this case to "_doc" internally, so this is what we need to check. I will try to make the comment clearer here.

Reply:
> To me it seems this gets converted in this case to "_doc" internally, so this is what we need to check.

I'm not sure why the custom type would get converted to `_doc` internally, I'm not aware of anywhere we would do that. I'll take some time to step through this test to try to understand what could be happening.

Reply:
It doesn't get converted. But we (need to) use a typeless template when putting it on a "new" cluster, because we cannot add a custom type name on >7.0 anymore. See Line 881, (isRunningAgainstAncientCluster will be false here when we are running agains 7.0 or later versions).

Reply:
Sorry, I didn't make this part clear after my last update. I don't think the comment in the code talks about "converting" but I certainly do im my above comment. Sorry if this was confusing, I didn't read the test that well back then.

Reply:
No problem, I think I now understand the intention of the test. I wonder if we could take the following simpler approach: https://github.com/jtibshirani/elasticsearch/commit/58dfb06729c9316933a543430119cce779b1a06e. I think the approach to handling 6.x versions in this was changed when we cut the 7.0 branch, but we never ended up removing these lines, which were part of the older approach.

Reply:
Good idea, I pushed an other commit.